crates/ty_python_semantic/resources/mdtest/attributes.md

```diff
@@ -1628,6 +1628,70 @@ date.year = 2025
 date.tz = "UTC"
 ```
 
+### Setting attributes on unions
+
+Setting attributes on unions where all elements of the union have the attribute is acceptable
+
+```py
+from typing import Union
+
+class A:
+    x: int
+
+class B:
+    x: int
+
+C = Union[A, B]
+
+a: C = A()
+a.x = 42
+```
+
+Setting attributes on unions where any element of the union does not have the attribute reports
+possibly unbound
+
+```py
+from typing import Union
+
+class A:
+    pass
+
+class B:
+    x: int
+
+C = Union[A, B]
+
+a: C = A()
+
+# instead of unresolved-attribute, this should report possibly-unbound-attribute
+# TODO: error: [possibly-unbound-attribute]
+# error: [unresolved-attribute]
+a.x = 42
```

Comment:
mypy:

`../test.py:12: error: Item "A" of "A | B" has no attribute "x"  [union-attr]`